fineract-provider/src/main/java/org/apache/fineract/accounting/journalentry/service/AccrualBasedAccountingProcessorForSavings.java

```diff
@@ -205,35 +212,45 @@ else if (savingsTransactionDTO.getTransactionType().isFeeDeduction() && savingsT
                             savingsProductId, paymentTypeId, savingsId, transactionId, transactionDate, overdraftAmount, isReversal,
                             penaltyPayments);
                     if (isPositive) {
+                        final ChargePaymentDTO chargePaymentDTO = penaltyPayments.get(0);
+                        AccrualAccountsForSavings accountTypeToBeDebited = AccrualAccountsForSavings.SAVINGS_CONTROL;
```

Comment:
I am not sure why to put this into a variable. The credit pair was inline in the method... :/

Reply:
Done! variable moved to be set before. This variable will be changed when we include Accrued Charges